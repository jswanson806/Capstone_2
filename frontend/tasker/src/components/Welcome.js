import React from 'react';

const Welcome = () => {
    return(
        <div>
            <h1>Welcome to Tasker!</h1>
            <p>Tasker is your personal marketplace for finding help to get things done. Sign up to post your open jobs or register as a Worker to get hired today!</p>
            <p>New here?</p>
            <a href='/signup'>Sign Up</a>
            <br></br>
            <p>Already registered?</p>
            <a href='/login'>Login</a>
        </div>
    );
}

export default Welcome;