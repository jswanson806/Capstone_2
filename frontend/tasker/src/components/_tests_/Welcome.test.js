import React from "react";
import { render } from "@testing-library/react";
import Welcome from "../Welcome.js";
import { MemoryRouter } from "react-router-dom";

describe('smoke and snapshot tests', () => {
    test('renders without crashing', () => {
        render(
            <MemoryRouter>
                <Welcome />
            </MemoryRouter>
        )
    })

    test('renders without crashing', () => {
        const {asFragment} = render(
            <MemoryRouter>
                <Welcome />
            </MemoryRouter>
        )

        expect(asFragment()).toMatchSnapshot();
    })
})