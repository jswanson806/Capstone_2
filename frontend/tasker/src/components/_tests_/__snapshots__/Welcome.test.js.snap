// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="welcome-card-container"
  >
    <div
      class="welcome-centered-card"
    >
      <h1>
        Welcome to Tasker!
      </h1>
      <h4>
        Tasker is your personal marketplace for finding help to get things done.
      </h4>
      <h6>
        Sign up as a 'User' to post your open jobs or as a 'Worker' to get hired today!
      </h6>
      <div
        class="welcome-buttons-container"
      >
        <h6>
          New here?
        </h6>
        <ul
          class="ml-auto nav"
        >
          <li
            class="mr-2 nav-item"
          >
            <a
              class="me-auto"
              href="/signup"
            >
              <button
                class="btn btn-info"
                data-testid="welcome-signup-button"
              >
                Sign Up
              </button>
            </a>
          </li>
        </ul>
        <h6>
          Already registered?
        </h6>
        <ul
          class="ml-auto nav"
        >
          <li
            class="mr-2 nav-item"
          >
            <a
              class="me-auto"
              href="/login"
            >
              <button
                class="welcome-login-button btn btn-warning"
                data-testid="welcome-login-button"
              >
                Login
              </button>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
